src/harness/reference_models/pre_iap_filtering/pre_iap_util.py

```diff
@@ -0,0 +1,186 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from shapely.geometry import shape, Point, Polygon
+from reference_models.geo import vincenty
+from sas_test_harness import generateCbsdReferenceId
+
+def purgeOverlappingGrants(cbsds, frequency_range):
+  """Removes Grants from CBSDs that overlap with the given frequency range.
+
+  Args:
+  cbsds: List of CbsdData objects. Grants are removed from the input.
+  frequency_range: A dictionary with keys 'lowFrequency' and 'highFrequency'
+  """
+  for cbsd in cbsds:
+    grants_to_purge = []
+    if cbsd['grants']:
```

Comment:
Although this works it is not the most pythonic or efficient: the remove operation will redo a search on the list to remove the actual grant you want to remove (so you have two loops going on in fact). 
More pythonic would be:
```
for cbsd in cbsds:
   if not cbsds['grants']:
      continue
   cbsds['grants'] = [grant for grant in cbsds['grants'] 
                       if not checkForOverlappingGrants(grant, frequency_range)]
```

Reply:
updated   as suggested in comment